### ISSUE TITLE
Added environment agnostics shasum tool, re-factor kindex_file_generator to utilize it [phab:D1305]

### ISSUE THREAD

**Issue by kythebot** [google/kythe#1691]
_Created by **craigbarber** at 2017-02-04 01:31:23_: 

This is necessary for some custom build rules which may run within environments which don't have the GNU tools installed which provide analogous functionality.

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-02-04 02:59:59_:

Please delete the blank separating the comment from the package spec, so that the godoc tool will know to use this as the package comment.

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-02-04 02:59:59_:

```
flag.Parse()
if flag.NArg() != 1 {
```

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-02-04 02:59:59_:

Suggestion:

```
	f, err := os.Open(flag.Arg(0))
	if err != nil {
		log.Fatal("Open: ", err)
	}
	defer f.Close()
	if _, err := io.Copy(hasher, f); err != nil {
		log.Fatal("Hashing failed: ", err)
	}
	fmt.Println(hex.EncodeToString(hasher.Sum(nil)))
```

That saves you having to read the whole file into memory. (That won't matter in most cases, but we do sometimes see input files over 100MiB).

**Comment by kythe** [comment]
_**craigbarber** wrote at 2017-02-07 17:18:06_:

  - incorporated code review feedback

**Comment by kythe** [comment]
_**fromberger** accepted this revision at 2017-02-07 17:20:25_

**Comment by kythe** [comment]
_**craigbarber** closed via commit 3bde8e8ccf0cd946eaa913b0aa5e04985d304618 at 2017-02-07 17:39:53_
